spring-cloud-dataflow-common/spring-cloud-dataflow-common-test-docker/src/test/java/org/springframework/cloud/dataflow/common/test/docker/compose/execution/DockerComposeTests.java

```diff
@@ -148,106 +142,90 @@ public void call_docker_compose_with_the_follow_flag_when_the_version_is_at_leas
     }
 
     @Test
-    public void not_throw_exception_when_down_fails_because_the_command_does_not_exist()
-            throws IOException, InterruptedException {
+    void notThrowExceptionWhenDownFailsBecauseTheCommandDoesNotExist() throws Exception {
         when(executedProcess.exitValue()).thenReturn(1);
         when(executedProcess.getInputStream()).thenReturn(toInputStream("No such command: down"));
         compose.down();
     }
 
     @Test
-    public void throw_exception_when_down_fails_for_a_reason_other_than_the_command_not_being_present()
-            throws IOException, InterruptedException {
+    void throwExceptionWhenDownFailsForAReasonOtherThanTheCommandNotBeingPresent() {
         when(executedProcess.exitValue()).thenReturn(1);
         when(executedProcess.getInputStream()).thenReturn(toInputStream(""));
-
-        exception.expect(DockerExecutionException.class);
-
-        compose.down();
+        assertThatExceptionOfType(DockerExecutionException.class)
+                .isThrownBy(() -> compose.down())
+                .withMessageStartingWith("'docker-compose down --volumes' returned exit code 1");
     }
 
     @Test
-    public void use_the_remove_volumes_flag_when_down_exists() throws IOException, InterruptedException {
+    void useTheRemoveVolumesFlagWhenDownExists() throws Exception {
         compose.down();
         verify(executor).execute("down", "--volumes");
     }
 
     @Test
-    public void parse_the_ps_output_on_ports() throws IOException, InterruptedException {
+    void parseThePsOutputOnPorts() throws Exception {
         Ports ports = compose.ports("db");
         verify(executor).execute("ps", "db");
         assertThat(ports, is(new Ports(new DockerPort("0.0.0.0", 7000, 7000))));
     }
 
     @Test
-    public void throw_illegal_state_exception_when_there_is_no_container_found_for_ports()
-            throws IOException, InterruptedException {
+    void throwIllegalStateExceptionWhereThereIsNoContainerFoundForPorts() {
         when(executedProcess.getInputStream()).thenReturn(toInputStream(""));
-        exception.expect(IllegalStateException.class);
-        exception.expectMessage("No container with name 'db' found");
-        compose.ports("db");
+        assertThatIllegalStateException()
+                .isThrownBy(() -> compose.ports("db"))
+                .withMessage("No container with name 'db' found");
     }
 
     @Test
-    public void pass_concatenated_arguments_to_executor_on_docker_compose_exec()
-            throws IOException, InterruptedException {
-        when(executedProcess.getInputStream()).thenReturn(toInputStream("docker-compose version 1.7.0rc1, build 1ad8866"));
-        compose.exec(options("-d"), "container_1", arguments("ls"));
-        verify(executor, times(1)).execute("exec", "-T", "-d", "container_1", "ls");
+    void failIfDockerComposeVersionIsPriorTo17OnDockerComposeExec() {
```

Comment:
Nitpick but it threw me reading `PriorTo17`.  In this case we could change the name to `failIfDockerComposeIsNotMinimumVersion`.   Then add a javadoc specifying that we only support 1.7 and above.

Reply:
Sure. But my guess is that this is proliferated all through this module/tests. I would like to delete all of this Docker compose stuff and use something off the shelf that probably now exists.

Reply:
The Javadoc should be somewhere in the DockerCompose code. I don't mind adding that simple thing but I would rather not go back through all this DC code and make it nice - I would prefer spending the time removing it ;)

Reply:
So this 1.7 is not a minimum version per-se. You can see elsewhere in the code that 1.7 is fine for most commands. It is only for `docker-compose exec` that this is a minimum version. We really don't care as we just use this for our tests and that is an ancient version.

Reply:
Renamed methods though - PTAL.